sql/scripts/SchemaSpy.sh

```diff
@@ -12,4 +8,35 @@ if [ ! -e "postgresql-9.4-1202.jdbc41.jar" ]; then
   wget https://jdbc.postgresql.org/download/postgresql-9.4-1202.jdbc41.jar;
 fi
 
-java -jar schemaSpy.jar -t pgsql -host $4 -s bookbrainz -db $3 -u $1 -p $2 -o _build -dp postgresql-9.4-1202.jdbc41.jar;
+while getopts "d:h:p:u:" opt; do
+  case $opt in
+    d)  DATABASE=$OPTARG;;
+    h)  HOST=$OPTARG;;
+    p)  PASSWORD=$OPTARG;;
+    u)  USERNAME=$OPTARG;;
+    *) echo "Error Occurred, Usage: SchemaSpy.sh -u [username] -p [password] -d [database] -h [host]" >&2
+       exit 1
+  esac
+done
+
+CMD="java -jar schemaSpy.jar -t pgsql -s bookbrainz "
+
+if [ ! -z "$HOST" ]
+then
+  CMD+="-host ${HOST} "
+fi
+
+if [ ! -z "$DATABASE" ]
+then
+  CMD+="-db ${DATABASE} "
+fi
+
+if [ ! -z "$USERNAME" ]
+then
+  CMD+="-u ${USERNAME} "
+fi
+
+CMD+="-p" "${PASSWORD}"
```

Comment:
Whenever I try to add a `-p mypassword` flag, I get an error `line 39: mypassword: command not found`.

That's because as it is, this line add "-p" to CMD and then _runs_ `${PASSWORD}` instead of adding it  to CMD as well.
You can turns this into `CMD+="-p '${PASSWORD}' "` to add the the flag and the quoted password—to avoid the IndexOutOfBoundsException— to CMD.